modules/fbx/fbx_parser/FBXParser.cpp

```diff
@@ -642,13 +668,18 @@ void ParseVectorDataArray(std::vector<Vector3> &out, const ElementPtr el) {
 
 		if (type != 'd' && type != 'f') {
 			print_error("expected float or double array (binary)" + String(token->StringContents().c_str()));
+			FBX_CORRUPT;
+			return;
 		}
 
 		std::vector<char> buff;
+		// count gets optimized away.
 		ReadBinaryDataArray(type, count, data, end, buff, el);
 
-		//ai_assert(data == end);
-		//ai_assert(buff.size() == count * (type == 'd' ? 8 : 4));
+		//		if (data != end || buff.size() == count * (type == 'd' ? 8 : 4)) {
+		//			FBX_CORRUPT;
+		//			return;
+		//		}
```

Comment:
If this is meant to be kept as commented out, would be good to at least fix style:
```suggestion
		//if (data != end || buff.size() == count * (type == 'd' ? 8 : 4)) {
		//	FBX_CORRUPT;
		//	return;
		//}
```
And have a comment that explains why it might be needed in the future.